airflow/providers/telegram/hooks/telegram.py

```diff
@@ -0,0 +1,154 @@
+        # or telegram_hook = TelegramHook(telegram_conn_id='telegram_default', chat_id='-1xxx')
+        # or telegram_hook = TelegramHook(token='xxx:xxx', chat_id='-1xxx')
+
+        # Call method from telegram bot client
+        telegram_hook.call(None', {"text": "message", "chat_id": "-1xxx"})
+        # or telegram_hook.call(None', {"text": "message"})
+
+    :param telegram_conn_id: connection that optionally has Telegram API token in the password field
+    :type telegram_conn_id: str
+    :param token: optional telegram API token
+    :type token: str
+    :param chat_id: optional chat_id of the telegram chat/channel/group
+    :type chat_id: str
+    """
+
+    def __init__(
+        self,
+        telegram_conn_id: Optional[str] = None,
+        token: Optional[str] = None,
+        chat_id: Optional[str] = None,
+    ) -> None:
+        super().__init__()
+        self.token = self.__get_token(token, telegram_conn_id)
+        self.chat_id = self.__get_chat_id(chat_id, telegram_conn_id)
+        self.connection = self.get_conn()
+
+    def get_conn(self) -> telegram.bot.Bot:
+        """
+        Returns the telegram bot client
+
+        :return: telegram bot client
+        :rtype: telegram.bot.Bot
+        """
+        return telegram.bot.Bot(token=self.token)
+
+    def __get_token(self, token: Optional[str], telegram_conn_id: str) -> str:
+        """
+        Returns the telegram API token
+
+        :param token: telegram API token
+        :type token: str
+        :param telegram_conn_id: telegram connection name
+        :type telegram_conn_id: str
+        :return: telegram API token
+        :rtype: str
+        """
+        if token is not None:
+            return token
+
+        if telegram_conn_id is not None:
+            conn = self.get_connection(telegram_conn_id)
+
+            if not conn.password:
+                raise AirflowException("Missing token(password) in Telegram connection")
+
+            return conn.password
+
+        raise AirflowException("Cannot get token: No valid Telegram connection supplied.")
+
+    def __get_chat_id(self, chat_id: Optional[str], telegram_conn_id: str) -> Optional[str]:
+        """
+        Returns the telegram chat ID for a chat/channel/group
+
+        :param chat_id: optional chat ID
+        :type chat_id: str
+        :param telegram_conn_id: telegram connection name
+        :type telegram_conn_id: str
+        :return: telegram chat ID
+        :rtype: str
+        """
+        if chat_id is not None:
+            return chat_id
+
+        if telegram_conn_id is not None:
+            conn = self.get_connection(telegram_conn_id)
+            return conn.host
+
+        return None
+
+    @tenacity.retry(
+        retry=tenacity.retry_if_exception_type(telegram.error.TelegramError),
+        stop=tenacity.stop_after_attempt(5),
+        wait=tenacity.wait_fixed(1),
+    )
+    def call(self, api_params: dict) -> None:
```

Comment:
IMHO this function name is misleading. From this name I would expect, hm... A call. :)
What do you think about `send_message` like the Telegram API method?

Reply:
I followed the name from slack hook but makes sense. changed it.